### ISSUE TITLE
Beta breaks my project: module resolution (?)

### ISSUE THREAD

**Issue by maghoff** [rust-lang/rust#53263]
The CI build of [my project Sausagewiki](https://github.com/maghoff/sausagewiki) has started failing with the beta compiler channel. See [the build log output](https://travis-ci.org/maghoff/sausagewiki/jobs/413920104#L678):

```
error[E0433]: failed to resolve. Use of undeclared type or module `articles`
   --> src/state.rs:326:22
    |
326 |             #[derive(Insertable)]
    |                      ^^^^^^^^^^ Use of undeclared type or module `articles`
```

It still builds with the stable compiler, so this unfortunately looks like a regression in rustc.

This project makes substantial use of code generation in `build.rs` and with procedural `derive` macros.

**Comment by petrochenkov** [comment]
This is most likely a consequence of https://github.com/rust-lang/rust/pull/51952 and the error originates in [diesel_derives](https://github.com/diesel-rs/diesel/blob/master/diesel_derives/src/insertable.rs).

It's strange however that this specific case is reported as an error and not as a lint warning (like many other warnings that can be seen in the log).

The issue is already fixed on diesel master, so perhaps switching to it will help as a workaround.

**Comment by maghoff** [comment]
Thanks for the feedback :)

I tried to see if I could get the problems to disappear with `[patch]`-ing the diesel dependencies in my Cargo.toml, but gave up after not managing to make it work. In other words, I was unable to apply the workaround. It doesn't matter much in the end as I have no particular use for it working in with the beta compiler.

However, I am very interested in my project not breaking with the stable compiler in the future :) I hope it will not be necessary for me to change anything for this to be the case.

**Comment by petrochenkov** [comment]
Fixed in https://github.com/rust-lang/rust/pull/53516

**Comment by maghoff** [comment]
I can verify that [my project now builds successfully with the beta channel](https://travis-ci.org/maghoff/sausagewiki/jobs/419024602).
